beaglebone-black/common/main_dma.c

```diff
@@ -242,22 +234,10 @@ static int dm_open(struct inode *inode, struct file *filp)
 
 			request_mem_region((unsigned long) mem_dev->base_addr, FPGA_MEM_SIZE, DEVICE_NAME);
 			mem_dev->virt_addr = ioremap_nocache(((unsigned long) mem_dev->base_addr), FPGA_MEM_SIZE);
-			mem_dev->dma_chan = edma_alloc_channel(EDMA_CHANNEL_ANY, dma_callback, NULL, EVENTQ_0);
-
-			if (mem_dev->dma_chan < 0) {
-				DBG_LOG("edma_alloc_channel failed for dma_ch, error: %d\n", mem_dev->dma_chan);
-
-				return mem_dev->dma_chan;
-			}
-
-			DBG_LOG("EDMA channel %d reserved\n", mem_dev->dma_chan);
-			mem_dev->dma_buf = (unsigned char *) dma_alloc_coherent(NULL, MAX_DMA_TRANSFER_IN_BYTES, &dmaphysbuf, 0);
 
-			if (mem_dev->dma_buf == NULL) {
-				DBG_LOG("failed to allocate DMA buffer\n");
-
-				return -ENOMEM;
-			}
+			result = logi_dma_init(mem_dev, &dmaphysbuf);
+			if (result)
```

Comment:
result is negative in case of an error! => should be changed to "if (result < 0)"